Snowflake.Data/Core/FileTransfer/SFFileTransferAgent.cs

```diff
@@ -667,155 +667,152 @@ private void initEncryptionMaterial()
-                    filePaths.AddRange(
-                        Directory.GetFiles(
-                            directoryName,
-                            fileName,
-                            SearchOption.TopDirectoryOnly));
-                }
             }
             else
             {
-                // No wild card, just make sure it's a file
-                FileAttributes attr = File.GetAttributes(location);
-                if (!attr.HasFlag(FileAttributes.Directory))
+                // No wildcard in the filename
+                foreach (var directory in foundDirectories)
                 {
-                    filePaths.Add(location);
-                }
-                else
-                {
-                    throw new FileNotFoundException(
-                        "Directories not supported, you need to provide a file path", location);
+                    var fullPath = Path.GetFullPath(directory + fileName);
+                    // Make sure it's a file
+                    var attr = File.GetAttributes(fullPath);
+                    if (attr.HasFlag(FileAttributes.Directory))
+                    {
+                        throw new FileNotFoundException(
+                            "Directories not supported, you need to provide a file path", fullPath);
+                    }
+                    filePaths.Add(fullPath);
                 }
             }
 
             if (Logger.IsDebugEnabled())
             {
-                Logger.Debug("Expand " + location + " into : \n");
-                foreach (string filepath in filePaths)
+                Logger.Debug("Expand " + location + " into: ");
+                foreach (var filepath in filePaths)
                 {
-                    Logger.Debug("\t" + filepath + "\n");
+                    Logger.Debug("\t" + filepath );
                 }
             }
 
             return filePaths;
         }
+        
+        /// <summary>
+        /// Expand the wildcards in the directory path to generate the list of directories to be searched for the files.
+        /// </summary>
+        /// <param name="directoryPath">The path to expand</param>
+        private static IEnumerable<string> ExpandDirectories(string directoryPath)
+        {
+            if (string.IsNullOrEmpty(directoryPath))
+            {
+                return new List<string> {Directory.GetCurrentDirectory() + Path.DirectorySeparatorChar};
+            }
+            if (!(directoryPath.Contains('?') || directoryPath.Contains('*')))
+            {
+                return new List<string> { Path.GetFullPath(directoryPath) + Path.DirectorySeparatorChar };
+            }
+            
+            var pathParts = directoryPath.Split(Path.DirectorySeparatorChar);
+            var resolvedPaths = new List<string>();
+
+            foreach (var part in pathParts)
+            {
+                if (part.Contains('?') || part.Contains('*'))
```

Comment:
here we can reuse method `pathContainsWildcards(part)`

Reply:
Done.